### DISCUSSION TITLE
POST uploadListingImage is malfunctioning

### DISCUSSION THREAD

**Discussion by yovasx2** [etsy/open-api#1311] [Q&A] [answered]
We have used POST uploadListingImage and we found some issues with it, it seems that the overwrite attribute is causing several things:
- If you hit GET getListingImages too fast after creating an image then you have eventual consistency, the first response does not contain the new image until several minutes after
When you include the **overwrite** parameter in the POST uploadListingImage some issues come:
- If the listing previously had an image with alt_text in the rank and you send a file (in a new request), then the alt_text is set to the previously deleted image:
  1. Create an image with alt_text
  2. Delete the image of the listing
  3. Send a new image file with new alt_text
  4. The new image is having the deleted image's alt_text  
- If the listing has 10 images (maximum) then a new request with a listing_image_id and overwrite triggers an error about the limit reached but then overwrite seems to be not taken into consideration
  1. Create a listing and attach 10 images to it
  2. Remove one image to get the listing_image_id (we are going to add this after getting the limit of images)
  3. Attach a new image (file) to the listing to reach the max again
  4. Send the post request using the referenced deleted image with overwrite true and rank 10
 - If you try to overwrite an image using reference with rank the original listing is sometimes **removed**

Expected:
The overwrite MUST replace images' parameters in the desired rank

**Comment by nebojsapetkovic94** [comment]
We have the same problem with alt_text. Is there any update on this issue?

**Comment by moshkh** [comment]
Bumping this, have a similar issue with alt text removal or updating.

**Comment by openapi-support** [comment, marked as answer]
Thank you for bringing this to our attention, the three bugs related to alt_text not being updated on overwrite, image slot silently deleted, and false max-images error on reposition have been fixed.
